git_creation_and_tracking/git_creation_and_tracking.md

```diff
@@ -0,0 +1,829 @@
+</div>
+***
+
+2. With each commit, we want to leave a short, descriptive, message describing the changes we make. Which of these would be the **best** commit message?
+
+[( )] "updates"
+[( )] "The Martian atmosphere is only 1% oxygen."
+[( )] "add a line that the Martian atmosphere is only 1% oxygen"
+[(X)] "add comments on Martian atmosphere"
+***
+<div class = "answer">
+The commit message should describe what this commit does. If you want to know the line by line changes you should use the `diff` command. The purpose of the message is so that you know what you changed without having to read all the individual changes, which is especially helpful if you edited many lines or files.
+</div>
+***
+
+## Telling Git not to track some files
+
+What if we have files that we do not want Git to track for us, like backup files created by our editor or intermediate files created during data analysis?
+
+<div class = "warning">
+It is likely that you will eventually share your repository.  This could be sharing it privately with a team, or publicly through [GitHub](github.com). When you do, everything that Git is tracking will become visible to everyone with access to the repository.  If there is an API key, password, sensitive data, or anything that requires any level of secrecy, make sure you tell Git to ignore it as soon as you create the file.
+
+Once information is committed and made public ("pushed" to GitHub), a record of it will persist even if you remove it in a later commit.
+</div>
+
+### Creating `.gitignore`
+
+Let's create a few dummy files to practice ignoring:
+
+```
+$ mkdir results
+$ touch a.dat b.dat c.dat results/a.out results/b.out
+```
+
+The `touch` command opens, saves, and closes the files without changing them. Since they didn't previously exist, `touch` is also creating them. Now if we check the status of our repository we see:
+
+```
+$ git status
+
+
+On branch main
+Untracked files:
+ (use "git add <file>..." to include in what will be committed)
+
+	a.dat
+	b.dat
+	c.dat
+	results/
+
+nothing added to commit but untracked files present (use "git add" to track)
+```
+
+In our scenario, these are files we don't want to track.  Putting these files under version control would be a waste of disk space. What's worse, having them all listed could distract us from changes that actually matter, so let's tell Git to ignore them.
+
+We do this by creating a file in the root directory of our project called `.gitignore`.   That dot at the beginning is intentional!  The file must be named this exactly for `.gitignore` to work properly.  Each line in this file is a rule for a type of file Git should ignore. Use your favorite text editor to make `.gitignore` contain the two lines:
+
+```
+*.dat
+results/
+```
+
+These patterns tell Git to ignore any file whose name ends in `.dat` and everything in the `results` directory. (If any of these files were already being tracked, Git would continue to track them.)
```

Comment:
Maybe a link to commonly used regex's? e.g. https://git-scm.com/docs/gitignore 

Edit: I see the SWC info on ignoring things is linked later. Could be sufficient!